Add and use Red-Commons library

Drapersniper opened the pull request:
### Description of the changes



### Have the changes in this PR been tested?
yes

Drapersniper committed:
update RC dep

Jackenmen commented on redbot/__init__.py:
```diff
@@ -16,7 +16,7 @@
 )
 
-from redbot._log import RedTraceLogger
+from red_commons.logging import maybe_update_logger_class
```
You can't do this, this will cause Red to fail to build (because it's an external dependency).

Drapersniper committed:
welp 100% tested

Drapersniper committed:
fix import

Drapersniper committed:
120% tested

Jackenmen committed:
Call _early_init even earlier

Not really in scope of this PR but the original was merged
before I could share any feedback.

Drapersniper committed:
explicitly import getLogger
